src/main/kotlin/pt/up/fe/ni/website/backend/annotations/validation/DateInterval.kt

```diff
@@ -0,0 +1,39 @@
+package pt.up.fe.ni.website.backend.annotations.validation
+
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import java.util.Date
+import kotlin.reflect.KClass
+import kotlin.reflect.full.memberProperties
+
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.RUNTIME)
+@Constraint(validatedBy = [DateIntervalValidator::class])
+@MustBeDocumented
+annotation class DateInterval(
```

Comment:
Perhaps change the annotation's name to avoid confusion with the Embeddable class. Something like `ValidDateInterval` maybe?